subgraphs/opensea-v2/src/mappings.ts

```diff
@@ -0,0 +1,480 @@
+// addrs[11] sell.target,
+// addrs[12] sell.staticTarget,
+// addrs[13] sell.paymentToken,
+
+// uints[0] buy.makerRelayerFee,
+// uints[1] buy.takerRelayerFee,
+// uints[2] buy.makerProtocolFee,
+// uints[3] buy.takerProtocolFee,
+// uints[4] buy.basePrice,
+// uints[5] buy.extra,
+// uints[6] buy.listingTime,
+// uints[7] buy.expirationTime,
+// uints[8] buy.salt
+// uints[9] sell.makerRelayerFee,
+// uints[10] sell.takerRelayerFee,
+// uints[11] sell.makerProtocolFee,
+// uints[12] sell.takerProtocolFee,
+// uints[13] sell.basePrice,
+// uints[14] sell.extra,
+// uints[15] sell.listingTime,
+// uints[16] sell.expirationTime,
+// uints[17] sell.salt
+
+// feeMethodsSidesKindsHowToCalls[0] buy.feeMethod
+// feeMethodsSidesKindsHowToCalls[1] buy.side
+// feeMethodsSidesKindsHowToCalls[2] buy.saleKind
+// feeMethodsSidesKindsHowToCalls[3] buy.howToCall
+// feeMethodsSidesKindsHowToCalls[4] sell.feeMethod
+// feeMethodsSidesKindsHowToCalls[5] sell.side
+// feeMethodsSidesKindsHowToCalls[6] sell.saleKind
+// feeMethodsSidesKindsHowToCalls[7] sell.howToCall
+
+export function handleMatch(call: AtomicMatch_Call): void {
+  // sellTarget is sell.target (addrs[11])
+  let sellTarget = call.inputs.addrs[11];
+  if (sellTarget.equals(WYVERN_ATOMICIZER_ADDRESS)) {
+    handleBundleSale(call);
+  } else {
+    handleSingleSale(call);
+  }
+}
+
+function handleSingleSale(call: AtomicMatch_Call): void {
+  let mergedCallData = guardedArrayReplace(
+    call.inputs.calldataBuy,
+    call.inputs.calldataSell,
+    call.inputs.replacementPatternBuy
+  );
+  if (!checkCallDataFunctionSelector(mergedCallData)) {
+    log.warning(
+      "[checkCallDataFunctionSelector] returned false, Method ID: {}, transaction hash: {}, target: {}",
+      [getFunctionSelector(mergedCallData), call.transaction.hash.toHexString(), call.inputs.addrs[11].toHexString()]
+    );
+    return;
+  }
+
+  let decodedTransferResult = decodeSingleNftData(mergedCallData);
+
+  let collectionAddr = decodedTransferResult.token.toHexString();
+  let tokenId = decodedTransferResult.tokenId;
+  let amount = decodedTransferResult.amount;
+
+  let saleKind = call.inputs.feeMethodsSidesKindsHowToCalls[6];
+  let strategy = getSaleStrategy(saleKind);
+
+  // paymentToken is buyOrder.paymentToken or SellOrder.payment token (addrs[6] or addrs[13])
+  let paymentToken = call.inputs.addrs[13];
+  // basePrice is buyOrder.basePrice or SellOrder.basePrice token (uints[4] or uints[13])
+  let basePrice = call.inputs.uints[13];
+  // TODO: calculate final price for dutch auction sales
+  let volumeETH = calcTradeVolumeETH(paymentToken, basePrice);
+  let priceETH = volumeETH;
+
+  // buyer is buyOrder.maker (addrs[1])
+  let buyer = call.inputs.addrs[1].toHexString();
+  // seller is sellOrder.maker (addrs[8])
+  let seller = call.inputs.addrs[8].toHexString();
+
+  // No event log index since this is a contract call
+  let tradeID = call.transaction.hash.toHexString();
+  let trade = new Trade(tradeID);
```

Comment:
what if multiple trades in a single tx?

Reply:
i think single sales only contain 1 trade in the transaction, but i would need to handle this case for bundle sales 
note: since this uses call handlers/there is no event log index

Reply:
changed to `{transactionHash}-{methodId}-{tokenId}`

Reply:
ah right. This is call handler so it's tricky.

What I meant is someone can group multiple function calls (i.e. multiple calls to atomicMatch) in the same tx. Or is that considered bundle sale?